framework/platform/bpl/uci/db/bpl_db_uci.h

```diff
@@ -0,0 +1,135 @@
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace beerocks {
+namespace bpl {
+namespace db {
+
+/****************************************************************************/
+/******************************* Definitions ********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************* Structures *********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************** Functions *********************************/
+/****************************************************************************/
+
+/**
+ * @brief Find if section exists.
+ * 
+ * @param[in] package_name name of the requested configuration file.
+ * @param[in] section_type type of the requested section.
+ * @param[in] section_name name of the requested section.
+ * @return true if section exists in db, false otherwise
+ */
+bool uci_section_exists(const std::string &package_name, const std::string &section_type,
+                        const std::string &section_name);
+
+/**
+ * @brief Add new named section.
+ * 
+ * @param[in] package_name name of the requested configuration file.
+ * @param[in] section_type type of the requested section.
+ * @param[in] section_name name of the requested section.
+ * @return true if section exists in db, false otherwise
+ */
+bool uci_add_section(const std::string &package_name, const std::string &section_type,
+                     const std::string &section_name);
+
+/**
+ * @brief Set values in section, updating as needed.
+ * An option set to an empty string will be removed from the section.
+ * An option that does not exist will be added to the section.
+ * An option that already exists will be overridden with the new value
+ * Options that exist and are not mentioned in @a params will remain unchanged.
+ * 
+ * @param[in] package_name name of the requested configuration file.
+ * @param[in] section_type type of the requested section.
+ * @param[in] section_name name of the requested section.
+ * @param[in] params unordered_map containing the params the requested section, to be set.
+ * @return true on success, false otherwise.
+ */
+bool uci_set_section(const std::string &package_name, const std::string &section_type,
+                     const std::string &section_name,
+                     const std::unordered_map<std::string, std::string> &options);
+
+/**
+ * @brief Get values in section.
+ * 
+ * @param[in] package_name name of the requested configuration file.
+ * @param[in] section_type type of the requested section.
+ * @param[in] section_name name of the requested section.
+ * @param[in/out] params unordered_map empty initalized map, that will contain found parameters. 
```

Comment:
All those `in/out` parameters are either in or out. On `set` apis it acts as `in` and on `get`  it's an `out`

Reply:
These `in/out` need to be initialized by the caller, therefore they are `in/out`

Reply:
https://stackoverflow.com/a/6900223   
`IN/OUT: An IN/OUT parameter usually indicates something where both the input and output values are meaningful. For instance, if you have a library function that fills a buffer, it may require you to pass a pointer to the buffer, along with another pointer indicating the length of the buffer. When the function returns, the second pointer may contain the actual number of bytes that have been written to the buffer.`